Common/src/gRPC/Services/GrpcApplicationsService.cs

```diff
@@ -79,4 +82,41 @@ public class GrpcApplicationsService : Applications.ApplicationsBase
              Total = tasks.totalCount,
            };
   }
+
+  [RequiresPermission(typeof(GrpcApplicationsService),
+                      nameof(CountTasksByStatus))]
+  public override async Task<CountTasksByStatusResponse> CountTasksByStatus(CountTasksByStatusRequest request,
+                                                                            ServerCallContext         context)
+  {
+    try
+    {
+      return new CountTasksByStatusResponse
+             {
+               Status =
+               {
+                 (await taskTable_.CountTasksAsync(data => data.Options.ApplicationName == request.Name && data.Options.ApplicationVersion == request.Version,
+                                                   context.CancellationToken)
+                                  .ConfigureAwait(false)).Select(count => new StatusCount
+                                                                          {
+                                                                            Status = count.Status,
+                                                                            Count  = count.Count,
+                                                                          }),
+               },
+             };
+    }
+    catch (ArmoniKException e)
+    {
+      logger_.LogWarning(e,
+                         "Error while counting tasks by applications");
+      throw new RpcException(new Status(StatusCode.Internal,
+                                        "Internal ArmoniK Exception, see application logs"));
+    }
+    catch (Exception e)
+    {
+      logger_.LogWarning(e,
+                         "Error while counting tasks by applications");
+      throw new RpcException(new Status(StatusCode.Unknown,
+                                        "Unknown Exception, see application logs"));
+    }
```

Comment:
Don't you need to catch some exceptions to make them user errors, instead of internal errors?

Reply:
The underlyning method does not throw exceptions. It will just return empty list most of the time.

Reply:
fair enough